src/app/mauro/query-builder.service.ts

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2022 University of Oxford
+and Health and Social Care Information Centre, also known as NHS Digital
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+import { Injectable } from '@angular/core';
+import {
+  CatalogueItemDomainType,
+  DataType,
+  Profile,
+} from '@maurodatamapper/mdm-resources';
+import { QueryBuilderConfig } from 'angular2-query-builder';
+import { catchError, forkJoin, map, Observable, of } from 'rxjs';
+import {
+  DataElementSearchResult,
+  DataRequestQueryPayload,
+  QueryExpression,
+} from '../data-explorer/data-explorer.types';
+import { ProfileService } from './profile.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class QueryBuilderService {
+  constructor(private profileService: ProfileService) {}
+
+  private getQueryBuilderDatatype(dataType?: DataType): Observable<Profile> {
```

Comment:
I'm surprised that `private` methods can go above `public` methods, I thought there was an eslint rule to enforce such ordering.

Reply:
I have moved private methods to the bottom of the class.